copier/copier.go

```diff
@@ -41,6 +41,74 @@ func init() {
 	reexec.Register(copierCommand, copierMain)
 }
 
+// extendedGlob calls filepath.Glob() on the passed-in patterns.  If there is a
+// "**" component in the pattern, filepath.Glob() will be called with the "**"
+// replaced with all of the subdirectories under that point, and the results
+// will be concatenated.
+func extendedGlob(pattern string) (matches []string, err error) {
+	subdirectories := func(dir string) []string {
+		var subdirectories []string
+		if err := filepath.WalkDir(dir, func(path string, d fs.DirEntry, _ error) error {
+			if d.IsDir() {
+				if rel, err := filepath.Rel(dir, path); err == nil {
+					subdirectories = append(subdirectories, rel)
+				}
+			}
+			return nil
+		}); err != nil {
+			subdirectories = []string{"."}
+		}
+		return subdirectories
+	}
+	expandPatterns := func(pattern string) []string {
+		patterns := []string{pattern}
+		dir, file := filepath.Split(pattern)
+		var rest string
+		for dir != "" && dir != string(os.PathSeparator) {
+			if file == "**" {
+				subdirectories := subdirectories(dir)
+				patterns := []string{}
+				for _, subdirectory := range subdirectories {
+					if rest == "" {
+						patterns = append(patterns, filepath.Join(dir, subdirectory))
+					} else {
+						patterns = append(patterns, filepath.Join(dir, subdirectory, rest))
+					}
+				}
+				return patterns
+			}
+			if rest == "" {
+				rest = file
+			} else {
+				rest = filepath.Join(file, rest)
+			}
+			if dir != "" && dir[len(dir)-1] == os.PathSeparator {
+				dir = dir[:len(dir)-1]
+			}
+			dir, file = filepath.Split(dir)
+		}
+		return patterns
+	}
+	patterns := []string{pattern}
+	i := 0
+	for i < len(patterns) {
+		expanded := expandPatterns(patterns[i])
+		if len(expanded) > 1 {
+			patterns = append(append(patterns[:i], expanded...), patterns[i+1:]...)
+		} else {
+			i++
```

Comment:
Why not use `break`?

Reply:
We don't know that `patterns[i+1]` (if we're not at the end of the slice) doesn't need to be expanded.